arangod/IResearch/IResearchView.cpp

```diff
@@ -2037,11 +2037,33 @@ void IResearchView::registerFlushCallback() {
     return;
   }
 
-  flush->registerCallback(this, [this]() noexcept {
-    return IResearchView::FlushTransactionPtr(
-      this,
-      [](arangodb::FlushTransaction*){} // empty deleter
+  auto viewSelf = self();
+
+  flush->registerCallback(this, [viewSelf]() noexcept {
```

Comment:
I think this function is not guaranteed to be `noexcept`, as it relies on copying or moving `std::function` does not throw exceptions, but that is not portable, as the standard doesn't guarantee it.